src/chain_sync/tipset_syncer.rs

```diff
@@ -882,7 +882,7 @@ async fn sync_headers_in_reverse<DB: Blockstore + Clone + Sync + Send + 'static,
         }
         // Attempt to load the parent tipset from local store
         if let Ok(tipset) = chain_store.tipset_from_keys(oldest_parent.parents()) {
-            parent_blocks.extend_from_slice(tipset.cids());
+            parent_blocks.extend(tipset.cids());
```

Comment:
Let's write an `IntoIterator<Item=Cid>` implementation:

```suggestion
            parent_blocks.extend(tipset);
```